### PR TITLE
347 Enable Linting for JSON Files

### LINKED ISSUE #347: Enable linting JSON files
The easiest way to enable linting from Go CLI is to start with JSON files. 

#### Acceptance criteria
1. Go CLI can lint JSON files
2. Requests are sent to ZALLY server
3. Violations are displayed
3. Few functional tests are added

Example how to unmarshal flexible JSON structures: https://play.golang.org/p/gfP6P4SmaC

### PR DESCRIPTION
**Pull request by netme**
Closes #347

### CODE REVIEW

File: cli-go/zally/readers/specs_reader.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package readers
+
+import (
+	"encoding/json"
+)
+
+// SpecsReader is an interface for all Swagger specification readers
+type SpecsReader interface {
+	read() (json.RawMessage, error)
```
Comment (maxim-tschumak) [inline cli-go/zally/readers/specs_reader.go]: I believe this interface method should be called `Read()` to match `func (j *JSONReader) Read() (json.RawMessage, error)`

Comment (netme) [inline cli-go/zally/readers/specs_reader.go]: Nice spot, I will fix that

File: cli-go/zally/commands/lint.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	var apiViolationsRequest domain.APIViolationsRequest
+	apiViolationsRequest.APIDefinition = &data
+	requestBody, err := json.MarshalIndent(apiViolationsRequest, "", "  ")
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	request := buildRequest("POST", fmt.Sprintf("%s/api-violations", baseURL), token, bytes.NewBuffer(requestBody))
+	request.Header.Add("Content-Type", "application/json")
```
Comment (maxim-tschumak) [inline cli-go/zally/commands/lint.go]: This header is used for all calls to the Zally server - should we move this to `buildRequest` function?

**Comment by netme** [comment]
@maxim-tschumak , could you please review it before it will become a huge PR? 

I've created follow-up tickets #361, #362 and #363

**Comment by maxim-tschumak** [comment]
:+1:

**Pull request merged**